sequencer/finalizer.go

```diff
@@ -580,6 +581,37 @@ func (f *finalizer) processTransaction(ctx context.Context, tx *TxTracker) (errW
 	} else if tx != nil && err == nil && !processBatchResponse.IsRomLevelError && len(processBatchResponse.Responses) == 0 {
 		err = fmt.Errorf("executor returned no errors and no responses for tx: %s", tx.HashStr)
 		f.halt(ctx, err)
+	} else if tx != nil && processBatchResponse.ExecutorError == runtime.ErrBalanceMismatch ||
```

Comment:
Reviewing again the code I think the correct place to check for these errors is in the handleProcessTransactionResult or handleProcessTransactionError function. We need to discuss/review this